llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1011,6 +1011,17 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
       TypeInfo.inferScalarType(R.getOperand(1)) ==
           TypeInfo.inferScalarType(R.getVPSingleValue()))
     return R.getVPSingleValue()->replaceAllUsesWith(R.getOperand(1));
+
+  // For i1 vp.merges produced by AnyOf reductions:
+  // vp.merge true, (or x, y), x, evl -> vp.merge y, true, x, evl
+  if (match(&R, m_Intrinsic<Intrinsic::vp_merge>(
+                    m_True(), m_c_BinaryOr(m_VPValue(X), m_VPValue(Y)),
+                    m_Deferred(X), m_VPValue())) &&
```

Comment:
This can match `vp.merge true, (or a, b), a, evl`.
Could it also match `vp.merge true, (or a, b), b, evl`?

Reply:
The binary or matcher is commutative so it should be able to

Reply:
I do something like 
```suggestion
  if (match(&R, m_Intrinsic<Intrinsic::vp_merge>(
                    m_True(), m_c_BinaryOr(m_VPValue(X), m_VPValue(Y)),
                    m_Deferred(Y), m_VPValue())) &&
```
and the test case vectorize-force-tail-with-evl-inloop-reduction.ll can not be detected.
Could you confirm it?

Reply:
Thanks for catching that, I see now that the commutative part won't make a difference because m_VPValue will always match.